src/Microsoft.Extensions.Configuration.Json/JsonConfigurationProvider.cs

```diff
@@ -52,18 +52,20 @@ public override void Load(Stream stream)
 
         private static string RetrieveErrorContext(JsonReaderException e, IEnumerable<string> fileContent)
         {
-            string errorLine;
+            string errorLine = null;
             if (e.LineNumber >= 2)
             {
                 var errorContext = fileContent.Skip(e.LineNumber - 2).Take(2).ToList();
-                errorLine = errorContext[0].Trim() + Environment.NewLine + errorContext[1].Trim();
+                if (errorContext.Count() >= 2)
```

Comment:
I think I get it: you are adding protection against a case in which `e.LineNumber` will be way off, right?